ouroboros-consensus-shelley/test/Test/Consensus/Shelley/Ledger/Golden.hs

```diff
@@ -54,11 +65,201 @@ tests = testGroup "Golden tests"
     , testCase "ApplyTxErr"     test_golden_ApplyTxErr
     , testCase "ConsensusState" test_golden_ConsensusState
     , testCase "LedgerState"    test_golden_LedgerState
-      -- TODO Query and result
+    , testCase "HeaderState"    test_golden_HeaderState
+    , testCase "ExtLedgerState" test_golden_ExtLedgerState
+    , testQueries
+    , testResults
     ]
 
 type Block = ShelleyBlock TPraosMockCrypto
 
+testQueries :: TestTree
+testQueries = testGroup "Queries"
+    [ testCase "LedgerTip"
+        $ test_golden_Query GetLedgerTip tipTerm
+    , testCase "EpochNo"
+        $ test_golden_Query GetEpochNo echoNoTerm
+    , testCase "NonMyopicMemberRewards"
+        $ test_golden_Query queryNonMyopicMemberRewards nonMyopicMemberRewardsTerm
+    , testCase "CurrentPParams"
+        $ test_golden_Query GetCurrentPParams currentPParamsTerm
+    , testCase "ProposedPParamsUpdates"
+        $ test_golden_Query GetProposedPParamsUpdates proposedPParamsUpdatesTerm
+    , testCase "StakeDistribution"
+        $ test_golden_Query GetStakeDistribution stakeDistributionTerm
+    ]
+  where
+    tipTerm, echoNoTerm, nonMyopicMemberRewardsTerm, currentPParamsTerm :: FlatTerm
+    proposedPParamsUpdatesTerm, stakeDistributionTerm :: FlatTerm
+
+    tipTerm =
+      [ TkListLen 1
+      , TkInt 0
+      ]
+
+    echoNoTerm =
+      [ TkListLen 1
+      , TkInt 1
+      ]
+
+    nonMyopicMemberRewardsTerm =
+      [ TkListLen 2
+      , TkInt 2
+      , TkTag 258
+      , TkListLen 1
+      , TkListLen 2
+      , TkInt 0
+      , TkBytes "\147\184\133\173"
+      ]
+
+    currentPParamsTerm =
+      [ TkListLen 1
+      , TkInt 3
+      ]
+
+    proposedPParamsUpdatesTerm =
+      [ TkListLen 1
+      , TkInt 4
+      ]
+
+    stakeDistributionTerm =
+      [ TkListLen 1
+      , TkInt 5
+      ]
+
+    queryNonMyopicMemberRewards
+      :: Query Block (NonMyopicMemberRewards TPraosMockCrypto)
+    queryNonMyopicMemberRewards =
+      GetNonMyopicMemberRewards $ Set.singleton $
+        (STS.KeyHashObj . SL.hashKey . SL.vKey) $ SL.KeyPair 0 0
```

Comment:
Actually it may be worth adding another element in the set with `ScriptHashObj`. I couldn't find any golden tests with this constructor.

Reply:
Good idea, go ahead.